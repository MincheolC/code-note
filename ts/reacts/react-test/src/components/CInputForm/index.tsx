import React, { useState } from 'react'

interface CInputFormProps {
  onSubmit: (e: React.FormEvent) => void;
}

const CInputForm: React.FC<CInputFormProps> = ({ onSubmit }) => {
  const [text, setText] = useState<string>('');
  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => setText(e.target.value);

  return (
    <form onSubmit={onSubmit}>
      <input value={text} onChange={onChange} />
      <button type='submit'>저장</button>
    </form>
  )
}

export default CInputForm