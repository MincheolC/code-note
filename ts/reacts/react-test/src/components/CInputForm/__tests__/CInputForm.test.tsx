import React from 'react';
import renderer, { ReactTestRenderer, ReactTestRendererJSON } from 'react-test-renderer';
import {cleanup, fireEvent, render} from '@testing-library/react';
import CInputForm from '../';

afterEach(cleanup);

test('CInputForm render properly', () => {
  const onSubmit = jest.fn();
  const component: ReactTestRenderer = renderer.create(
    <CInputForm onSubmit={onSubmit}/>,
  );

  let tree = component.toJSON() as ReactTestRendererJSON;
  expect(tree).toMatchSnapshot();
});

it('CInputForm pass input text to onSubmit', () => {
  const onSubmit = jest.fn();
  const inputValue = 'test';

  const {getByText, getByLabelText, queryByLabelText} = render(
    <CInputForm onSubmit={onSubmit} />,
  );

  fireEvent.change(getByLabelText(/todo/i), { target: { value: inputValue }});
  fireEvent.click(getByText(/저장/i));

  expect(queryByLabelText(/todo/i)).toHaveValue('');
  expect(onSubmit).toBeCalled();
  expect(onSubmit.mock.calls).toEqual([['test']]);
});