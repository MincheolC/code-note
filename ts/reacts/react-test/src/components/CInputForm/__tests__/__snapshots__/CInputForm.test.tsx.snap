// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CInputForm render properly 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="todo"
  >
    할 일
  </label>
  <input
    aria-label="todo"
    name="todo"
    onChange={[Function]}
    value=""
  />
  <button
    type="submit"
  >
    저장
  </button>
</form>
`;
